aderyn_core/src/detect/low/const_multiple_definitions.rs

```diff
@@ -0,0 +1,114 @@
+use std::collections::hash_map::Entry;
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::error::Error;
+
+use crate::ast::{NodeID, VariableDeclaration};
+
+use crate::capture;
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct ConstantDefinedMultipleTimesDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for ConstantDefinedMultipleTimesDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        // capture!(self, context, item);
+
+        let mut vars_by_name: HashMap<String, Vec<VariableDeclaration>> = HashMap::new();
+
+        for var_decl in context
+            .variable_declarations()
+            .into_iter()
+            .filter(|v| v.state_variable && v.constant)
+        {
+            match vars_by_name.entry(var_decl.name.clone()) {
+                Entry::Occupied(mut o) => {
+                    o.get_mut().push(var_decl.to_owned());
+                }
+                Entry::Vacant(e) => {
+                    e.insert(vec![var_decl.to_owned()]);
+                }
+            }
+        }
+
+        for (_, const_potentially_defined_multiple_times) in vars_by_name.into_iter() {
+            let mut locations: HashSet<_> = HashSet::new();
+            for c in &const_potentially_defined_multiple_times {
+                locations.insert(context.get_node_sort_key(&c.into()));
+            }
```

Comment:
I find that in the current setup, there can be duplicate nodes (2 different node IDs pointing to the same node)

That's why had to do this.

Reply:
@alexroan I think this is because foundry's out files have ASTs of the same contract which are coming from different compilation groups